crypto/fipsmodule/cipher/e_aes.c

```diff
@@ -2181,3 +2182,151 @@ DEFINE_METHOD_FUNCTION(EVP_CIPHER, EVP_xaes_256_gcm_kc) {
     out->cleanup = aes_gcm_cleanup;
     out->ctrl = xaes_256_gcm_kc_ctrl;
 }
+
+// -----------------------------------------------------------------------------
+// ------------------ EVP_AEAD XAES-256-GCM With Key Commitment ----------------
+// ----------- Reference: https://eprint.iacr.org/2025/758.pdf#page=6 ----------
+// -----------------------------------------------------------------------------
+static int aead_xaes_256_gcm_kc_init(EVP_AEAD_CTX *ctx, const uint8_t *key,
+                            size_t key_len, size_t requested_tag_len) {
+    // Max tag and key commitment length: 16 + 32 bytes
+    if(requested_tag_len > EVP_AEAD_AES_GCM_TAG_LEN + XAES_256_GCM_KEY_COMMIT_SIZE) {
+        OPENSSL_PUT_ERROR(CIPHER, CIPHER_R_UNSUPPORTED_TAG_SIZE);
+        return 0;
+    }
+
+    AEAD_XAES_256_GCM_CTX *xaes_ctx = (AEAD_XAES_256_GCM_CTX*)&ctx->state;
+
+    xaes_256_gcm_ctx_init(&xaes_ctx->xaes_key, xaes_ctx->k1, key);
+
+    // requested_tag_len = 0 means using the default tag length of XAES_256_GCM_KC
+    ctx->tag_len = (requested_tag_len > 0) ? requested_tag_len : 
+                   (EVP_AEAD_AES_GCM_TAG_LEN + XAES_256_GCM_KEY_COMMIT_SIZE);
+
+    return 1;
+}
+
+static int aead_xaes_256_gcm_key_commit_seal_scatter(
+    const EVP_AEAD_CTX *ctx, uint8_t *out,
+    uint8_t *out_tag, size_t *out_tag_len,
+    size_t max_out_tag_len,
+    const uint8_t *nonce, const size_t nonce_len,
+    const uint8_t *in, const size_t in_len,
+    const uint8_t *extra_in,
+    const size_t extra_in_len, const uint8_t *ad,
+    const size_t ad_len) {
+    /* Assume tag includes MAC concatenated with key commitment
+     * i.e., TAG = MAC (16-byte) || KEY_COMMITMENT (32-byte) 
+     * Therefore, truncating tag happens to key commitment first
+     * We consider the following possible cases of tag_len: 
+     * tag_len > 16 : tag includes 16-byte MAC tag and (tag_len - 16)-byte key commitment   
+     * tag_len <= 16: tag includes only (tag_len)-byte MAC tag and 0-byte key commitment */
```

Comment:
Did we decide in the design that we want to support this case or we error out as this would not be a key commitment cipher so someone calling it doesn't think key commitment exists in the tag?